lib/spack/spack/compilers/cce.py

```diff
@@ -65,6 +65,12 @@ def cxx11_flag(self):
             return '-std=c++11'
         return "-h std=c++11"
 
+    @property
+    def cxx14_flag(self):
+        if self.is_clang_based:
+            return '-std=c++14'
+        return "-h std=c++14"
```

Comment:
```suggestion
        raise UnsupportedCompilerFlag(self, 'the c++14 standard', 'cxx14_flag')
```

If it's not clang based it applies to the classic cray compiler, which likely does not support c++14, but I'm not 100% sure.

Reply:
I've tested pre-Clang CCE 8.7.9
```
> module load cdt/19.03
Switching to atp/2.1.3.
Switching to cce/8.7.9.
Switching to cray-libsci/18.12.1.
Switching to cray-mpich/7.7.6.
Switching to craype/2.5.18.
Switching to modules/3.2.11.1.
Switching to perftools-base/7.0.6.
Switching to pmi/5.0.14.
> CC -h std=c++14 hello.omp.cpp
> ./a.out
Hello World... from thread = 0
```

That's the oldest CCE I have access to right now.. So c++14 should be safe for sites with up-to-date classic CCE

Reply:
Ok, nice, finally someone with access to ancient compilers to fill in the gaps :p